abrechnung/database/__init__.py

```diff
@@ -11,13 +12,38 @@
 from abrechnung.config import Config
 from . import revisions
 
+logger = logging.getLogger(__name__)
+
 
 async def db_connect(
-    username: str, password: str, database: str, host: str, port: int = 5432
+        username: str, password: str, database: str, host: str, port: int = 5432
 ) -> Pool:
     """
     get a connection pool to the database
     """
+
+    # since marshmallow can't model a "one arg implies all"-relation, we need to warn here
+    if not host or os.path.isdir(host):
+        if username or password:
+            logger.warning("Username and/or password specified but no remote host therefore using socket "
+                           "authentication. I am ignoring these settings since we don't need them.")
+        if os.getenv("PGHOST") or os.getenv("PGPORT"):
+            # asyncpg can read the PGHOST env variable. We don't want that.
+            logger.warning("We do not support setting the PGHOST or PGPORT environment variable and therefore will "
+                           "ignore it. Consider specifying the hostname in the config file.")
+
+        if os.environ.get("PGHOST"):
+            del os.environ["PGHOST"]
+        if os.environ.get("PGPORT"):
+            del os.environ["PGPORT"]
+
+        if not host:
+            # let asyncpg figure out which socket to use
+            return await asyncpg.create_pool(database=database, max_size=100)
+        else:
+            # manually set folder in which the socket resides
+            return await asyncpg.create_pool(host=host, port=port, database=database, max_size=100)
```

Comment:
to avoid duplication you can create an argument dict with the args both variants have (expecially the max_size), and then optionally add host and port if host is not None.
```py
pool_args = dict(database=database, max_size=100)
if host:
    pool_args['host'] = host
    pool_args['port'] = port
return await asyncpg.create_pool(**pool_args)
```

In a similar fashion this should be merged with the "full" create pool call at the end of this function, so there's only one such "central" invocation